python/lsst/ip/isr/crosstalk.py

```diff
@@ -23,17 +23,441 @@
+
+        calib.updateMetadata()
+        return calib
+
+    def toDict(self):
+        """Return a dictionary containing the calibration properties.
+
+        The dictionary should be able to be round-tripped through
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+
+        outDict = {}
+        metadata = self.getMetadata()
+        outDict['metadata'] = metadata
+
+        outDict['hasCrosstalk'] = self.hasCrosstalk
+        outDict['nAmp'] = self.nAmp
+        outDict['crosstalkShape'] = self.crosstalkShape
+
+        ctLength = self.nAmp*self.nAmp
+        outDict['coeffs'] = self.coeffs.reshape(ctLength).tolist()
+
+        if self.coeffErr is not None:
+            outDict['coeffErr'] = self.coeffErr.reshape(ctLength).tolist()
+        if self.coeffNum is not None:
+            outDict['coeffNum'] = self.coeffNum.reshape(ctLength).tolist()
+
+        if self.interChip:
+            outDict['interChip'] = dict()
+            for detector in self.interChip:
+                outDict['interChip'][detector] = self.interChip[detector].reshape(ctLength).tolist()
+
+        return outDict
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct calibration from a list of tables.
+
+        This method uses the `fromDict` method to create the
+        calibration, after constructing an appropriate dictionary from
+        the input tables.
+
+        Parameters
+        ----------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables to use to construct the crosstalk
+            calibration.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.CrosstalkCalib`
+            The calibration defined in the tables.
+
+        """
+        coeffTable = tableList[0]
+
+        metadata = coeffTable.meta
+        inDict = dict()
+        inDict['metadata'] = metadata
+        inDict['hasCrosstalk'] = metadata['HAS_CROSSTALK']
+        inDict['nAmp'] = metadata['NAMP']
+
+        inDict['coeffs'] = coeffTable['CT_COEFFS']
+        if len(tableList) > 1:
+            inDict['interChip'] = dict()
+            interChipTable = tableList[1]
+            for record in interChipTable:
+                inDict['interChip'][record['IC_SOURCE_DET']] = record['IC_COEFFS']
+
+        return cls().fromDict(inDict)
+
+    def toTable(self):
+        """Construct a list of tables containing the information in this calibration.
+
+        The list of tables should create an identical calibration
+        after being passed to this class's fromTable method.
+
+        Returns
+        -------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables containing the crosstalk calibration
+            information.
+
+        """
+        tableList = []
+        self.updateMetadata()
+        catalog = Table([{'CT_COEFFS': self.coeffs.reshape(self.nAmp*self.nAmp)}])
```

Comment:
I agree with you in principle for using astropy Table instead of afwTable, but it's worth asking if we sacrifice anything in speed by using astropy?

Reply:
The astropy implementation is faster than afwTable, for reasons I don't completely remember.  I believe it has something to do with how afwTable stores the data to allow for row inserts.  My short timing test for DM-24537 confirmed this:  https://github.com/lsst/ip_isr/pull/135#issuecomment-620847931

Reply:
Yes, we need to switch all our curated calibrations over to astropy table. It's faster and @TallJimbo indicated that we should only be using afwTable if we needed to pass the tables to C++ (which we don't).